### PR TITLE
Sync with upstream

### PR DESCRIPTION
**Pull request by saghul**
Ignore the commits, there is a lot of noise due to cherry-picking by oney/master from jitsi/master. Just check the diff please.

### CODE REVIEW

File: RTCPeerConnection.js [hunk 1]
```diff
@@ -206,6 +200,9 @@ export default class RTCPeerConnection extends EventTarget(PEER_CONNECTION_EVENT
             // general, the stress is on being faster to pass through the React
             // Native bridge which is a bottleneck that tends to be visible in
             // the UI when there is congestion involving UI-related passing.
+            if (Array.isArray(stats) && stats.length === 1 && typeof stats[0] === 'string') {
```
Comment (paweldomas) [inline RTCPeerConnection.js:203]: why is that necessary ? is it possible to have something else than a JSON string currently ? if not then maybe we should check if it's a string and throw an error

**Review by paweldomas — APPROVED** [review]
LGTM - just one question inline

File: RTCPeerConnection.js (hunk 1 above)
Comment (saghul) [reply RTCPeerConnection.js:203]: Good catch. I believe iOS returns (or used to) an array of a single element instead of the actual stuff. I plan to switch to the non-deprecated stats API and I'll take care of this then.

Comment (paweldomas) [reply RTCPeerConnection.js:203]: so we want this as-is for now ?

Comment (paweldomas) [reply RTCPeerConnection.js:203]: btw. there are some conflicts

Comment (saghul) [reply RTCPeerConnection.js:203]: I'd say so, yeah. What conflicts?
